File: app/src/main/kotlin/com/mobileappdev/homeworkplanner/ClassListActivity.java
package com.mobileappdev.homeworkplanner;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;

import com.example.myapplication.R;

public class ClassListActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_class_list);

        FragmentManager fm = getSupportFragmentManager();
        Fragment fragment = fm.findFragmentById(R.id.class_list_fragment_container);

        if (fragment == null){
            fragment = new ClassListFragment();
            fm.beginTransaction().add(R.id.class_list_fragment_container, fragment).commit();
        }
    }
}

File: app/src/main/kotlin/com/mobileappdev/homeworkplanner/ClassListActivity.kt
package com.mobileappdev.homeworkplanner


import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.example.myapplication.R


class ClassListActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_fragment)

        val fm = supportFragmentManager
        var fragment = fm.findFragmentById(R.id.fragment_container)

        if (fragment == null) {
            fragment = ClassListFragment()
            fm.beginTransaction().add(R.id.fragment_container, fragment).commit()
            Log.d("lifecycle", "onCreate invoked")
        }
    }

}